src/encoded/types/individual.py

```diff
@@ -118,3 +134,121 @@ class IndividualZebrafish(Individual):
     item_type = 'individual_zebrafish'
     schema = load_schema('encoded:schemas/individual_zebrafish.json')
     embedded_list = Individual.embedded_list + ['zebrafish_vendor.name']
+
+
+# validator for individual relations - same species
+def validate_individual_relations(context, request):
+    '''Validates that individual relations are within the same species,
+    limited to two relations per individual (max 1 paternal and 1 maternal),
+    not self relations, and unique (no duplicate relations).
+    '''
+    #import pdb; pdb.set_trace()
+    data = request.json
+    organism = data.get('organism')
+    print(organism)
+    if organism is None:
+        return
+    get_organism = get_item_if_you_can(request, organism, 'organisms')
+    if get_organism:
+        organism_uuid = get_organism.get('uuid')
+    else:
+        return
+
+    related_individuals = data.get('individual_relation')  # a list of dicts
+    print(related_individuals)
+    if related_individuals is None:
+        return
+    any_failures = False
+
+    # Max 2 parents per individual
+    if len(related_individuals) > 2:
+        request.errors.add(
+            'body', 'Individual relation: too many parents',
+            'An individual cannot have more than two parents'
+        )
+        any_failures = True
+
+    relations_counter = {}
+    relations_unique = {}
+    for a_related_individual in related_individuals:
+        parent = a_related_individual.get('individual')
+        print(parent)
+        parent_props = get_item_if_you_can(request, parent, 'individuals', frame='raw')
+        print(parent_props)
+        if parent_props:
+            parent_organism = parent_props.get('organism')
+            parent_uuid = parent_props.get('uuid')
```

Comment:
This parent_uuid doesn't get created if parent_props aren't found, which I think is causing some problems. You can either indent all the code below another level or add an else clause.

Don't forget to take out the frame=raw a few lines above, as well.

Reply:
Does it make sense to keep the if clause at all? 
Given a parent individual, the get_item_if_you_can() will always return the item, unless something is very wrong. Or would this be a way to test if the wrong item has been added as parent (e.g. enzyme, instead of individual)?